addons/web_editor/controllers/main.py

```diff
@@ -607,7 +607,17 @@ def shape(self, module, filename, **kwargs):
                     or attachment.type != 'binary'
                     or not attachment.public
                     or not attachment.url.startswith(request.httprequest.path)):
-                raise werkzeug.exceptions.NotFound()
+                # Fallback to URL lookup to allow using shapes that were imported from data files.
+                attachment = request.env['ir.attachment'].sudo().search([
+                    '|',
+                        ('url', '=like', request.httprequest.path),
+                        # Also find files that have their dynamic colors listed.
+                        ('url', '=like', '%s?%%' % request.httprequest.path),
```

Comment:
Ok, but why would you list them in the URL field of the attachment? :) It seems to me it's just two lines of code (3 with the comment) just to be more tolerant of data definition... I'd always prefer to be more strict for the same feature if it means less code.

So unless this unlocks a feature I don't understand, I'd remove this :wink:

Reply:
Removed.
I got confused with the fact that undraw media are stored with dynamic color parameters inside the attachment URL, and would thus not be found without this extra domain part. But indeed those will not be shapes.